### ISSUE TITLE
Refactor -> Convert parameters to destructured object, error "Could not apply refactoring"

### ISSUE THREAD

**Issue by sliftist** [microsoft/TypeScript#38028]
**TypeScript Version:**  3.8.3 (vscode 1.44.2)

**Search Terms:**
`Could not apply refactoring`, `Convert parameters to destructured object`

**Code**

```ts
function test(x: number, y: number) { }
type T = ReturnType<typeof test>;
test(5, 1);
```

Try to use `Refactor -> convert parameters to destructured object` on test.

**Expected behavior:**
Refactors `test` to use a destructured object as its argument.

**Actual behavior:**
Error, `Could not apply refactoring`

**NOTE:**
The error is triggered by `typeof test`. I'm not sure if this problem is even tractable (considering the type operations on `typeof test` might pull apart the arguments, and so could be impossible to update), however in this specific case I only use the `ReturnType`, so the conversion is simple.

Perhaps there should just be a warning if `typeof` is used on the function, or a more specific error with a message describing why the function can't be refactored.

**Comment by RyanCavanaugh** [comment]
We're separately working on having refactors show why they failed in a user-facing way.

I don't think we'll special case `ReturnType<typeof f>` for refactoring purposes; we have a very large amount of other higher priority work that needs taking care of first

**Comment by danieltroger** [comment]
Hey man it still doesn't work, it just says "Could not apply refactoring"

**Comment by r-dev-limited** [comment]
still doesnt work

**Comment by Pavel-Retunskih** [comment]
Hi from 2024, still doesnt work

**Comment by robbie-hinman** [comment]
Still not working

**Comment by DayKev** [comment]
> We're separately working on having refactors show why they failed in a user-facing way.

@RyanCavanaugh So... any progress on this? Tried to use this refactoring option today and just got a generic (and very unhelpful) "Could not apply refactoring" error message.

**Comment by RyanCavanaugh** [comment]
We're likely not going to port this refactor to TS7
